src/cswinrt/main.cpp

```diff
@@ -188,6 +192,7 @@ Where <spec> is one or more of:
                                 continue;
                             }
                             auto guard{ w.push_generic_params(type.GenericParam()) };
+                            auto guard1{ helperWriter.push_generic_params(type.GenericParam()) };
```

Comment:
I don't think helperWriter is needed - you're only ever calling write_temp

Reply:
I believe we do need this because it is tied to the generic params that we set here. The write_temp functions uses the generic_params that we push to the helperWriter to write the type name and so on

Reply:
but w is pushing those very same generic params